src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,289 @@
+        return new NumberPayloadGeneratorUIPanel();
+    }
+
+    public static class NumberPayloadGeneratorUI implements PayloadGeneratorUI<DefaultPayload, NumberPayloadGenerator> {
+
+        NumberPayloadGenerator numberPayloadGenerator;
+
+        public NumberPayloadGeneratorUI(NumberPayloadGenerator generator) {
+            this.numberPayloadGenerator = generator;
+        }
+
+        @Override
+        public Class<? extends NumberPayloadGenerator> getPayloadGeneratorClass() {
+            return NumberPayloadGenerator.class;
+        }
+
+        @Override
+        public String getName() {
+            return PAYLOAD_GENERATOR_NAME;
+        }
+
+        @Override
+        public String getDescription() {
+            return PAYLOAD_GENERATOR_NAME;
+        }
+
+        @Override
+        public long getNumberOfPayloads() {
+            try {
+                return numberPayloadGenerator.getNumberOfPayloads();
+            } catch (Exception e) {
+                LOGGER.warn("Failed to obtain number of payloads", e);
+            }
+            return PayloadGenerator.UNKNOWN_NUMBER_OF_PAYLOADS;
+        }
+
+        @Override
+        public NumberPayloadGenerator getPayloadGenerator() {
+            return numberPayloadGenerator;
+        }
+
+        @Override
+        public PayloadGeneratorUI<DefaultPayload, NumberPayloadGenerator> copy() {
+            return this;
+        }
+        
+    }
+
+    public static class NumberPayloadGeneratorUIPanel extends AbstractPersistentPayloadGeneratorUIPanel<DefaultPayload, NumberPayloadGenerator, NumberPayloadGeneratorUI> {
+
+        private static final String PAYLOADS_PREVIEW_FIELD_LABEL = Constant.messages.getString("fuzz.payloads.generator.regex.payloadsPreview.label");
+        private static final String PAYLOADS_PREVIEW_GENERATE_FIELD_LABEL = Constant.messages.getString("fuzz.payloads.generator.regex.payloadsPreviewGenerate.label");
+
+        private static final int MAX_NUMBER_PAYLOADS_PREVIEW = 250;
+
+        private JPanel fieldsPanel;
+        private JTextArea payloadsPreviewTextArea;
+        private JButton payloadsPreviewGenerateButton;
+        private JTextField fromField, toField, stepField;
+        
+        private NumberPayloadGeneratorUI oldGenerator;
+
+        public NumberPayloadGeneratorUIPanel() {
+
+            fieldsPanel = new JPanel();
+
+            GroupLayout layout = new GroupLayout(fieldsPanel);
+            fieldsPanel.setLayout(layout);
+            layout.setAutoCreateGaps(true);
+
+            fromField = new JTextField();
+            toField = new JTextField();
+            stepField = new JTextField();
+
+            JLabel fromLabel = new JLabel("From:");
+            fromLabel.setLabelFor(fromField);
+
+            JLabel toLabel = new JLabel("To:");
+            toLabel.setLabelFor(toField);
+
+            JLabel stepLabel = new JLabel("Step:");
```

Comment:
This label should use an i18n string. (Should this be "Increment" for consistency?)